app/src/main/java/org/schabi/newpipe/fragments/list/channel/ChannelFragment.java

```diff
@@ -524,9 +529,15 @@ public void handleResult(@NonNull final ChannelInfo result) {
             playlistControlBinding.getRoot().setVisibility(View.GONE);
         }
 
+        channelContentNotSupported = false;
         for (final Throwable throwable : result.getErrors()) {
             if (throwable instanceof ContentNotSupportedException) {
-                showContentNotSupported();
+                /*
+                channelBinding might not be initialized when handleResult() is called
+                (e.g. after rotating the screen, https://github.com/TeamNewPipe/NewPipe/issues/6696)
+                showContentNotSupported() will be called later
+                 */
+                channelContentNotSupported = true;
```

Comment:
Uhm, and what if `onViewCreated` has already been called when this point is reached? Then `showContentNotSupported();` will never be called.

Reply:
I haven't encountered this situation before.

Another possible solution for this issue is to swap the order of these two lines. But idk if `handleResult()` will be called before `onViewCreated()`.
https://github.com/TeamNewPipe/NewPipe/blob/173b6c3f00df02e64151ee642323205400d412f2/app/src/main/java/org/schabi/newpipe/fragments/list/channel/ChannelFragment.java#L130-L133

Reply:
I would rename `showContentNotSupported` to `showContentNotSupportedIfNeeded` and have it behave this way:
```
if (!channelContentNotSupported || channelBinding == null) {
    return;
}
// all of the core currently in showContentNotSupported
```
Then it could be called both in `onViewCreated` (without the need for the `if`) and here just under `hannelContentNotSupported = true;`. This way we can be sure.